### ISSUE TITLE
ASD_POCS returns "no available GPUs that support cuda" in python

### ISSUE THREAD

**Issue by AnderBiguri** [CERN/TIGRE#277]
This is interesting, because I don't think it did before. 

Lack of CI means I have at some point introduced some bug or merged something without testing that make TV algorithms not work. 

This is a priority bug.

**Comment by AnderBiguri** [comment]
@genusn any help here, please?

I found out that this only happens if you don't give the gpuids to the algorithm. i.e.

```
out = algs.asd_pocs(proj,geo,angles,5) #fail
out = algs.asd_pocs(proj,geo,angles,5,gouids=gpuids) #succeeds

```

Im an trying to find the mistake, but I can't find it. 
The MATLAB side works.

**Comment by genusn** [comment]
I'll tackle it on the weekend.

**Comment by AnderBiguri** [comment]
@genusn no worries, I think I got it, and its certainly something that is my problem rather than your PR. 

The functions that call POCS_TV.cu are not in a separate file (as the ones that call tvdenoising.cu are). 
Simply, those functions do not have the:

```
if self.gpuids is None:
            self.gpuids = GpuIds()
```
lines. 

Found them, will fix them :) 

Apologies, this is something that should be different, these functions should probably be on a separate file (probably all of them together in a "regularization.py" file).
